merkledag/merkledag.go

```diff
@@ -394,56 +394,54 @@ func EnumerateChildren(ctx context.Context, ds LinkService, root *cid.Cid, visit
 var FetchGraphConcurrency = 8
 
 func EnumerateChildrenAsync(ctx context.Context, ds DAGService, c *cid.Cid, visit func(*cid.Cid) bool) error {
-	if !visit(c) {
-		return nil
-	}
-
-	root, err := ds.Get(ctx, c)
-	if err != nil {
-		return err
-	}
-
-	feed := make(chan node.Node)
-	out := make(chan *NodeOption)
+	feed := make(chan *cid.Cid)
+	out := make(chan []*cid.Cid)
 	done := make(chan struct{})
 
 	var setlk sync.Mutex
-
+	
+	errChan := make(chan error)
+	fetchersCtx, cancel := context.WithCancel(ctx)
```

Comment:
put a `defer cancel()` after this call and then remove the other call youre making to cancel